src/components/views/settings/ChangeAvatar.js

```diff
@@ -16,67 +16,65 @@ limitations under the License.
 
 import React from 'react';
 import PropTypes from 'prop-types';
-import createReactClass from 'create-react-class';
 import {MatrixClientPeg} from "../../../MatrixClientPeg";
 import * as sdk from '../../../index';
 import { _t } from '../../../languageHandler';
 
-export default createReactClass({
-    displayName: 'ChangeAvatar',
-    propTypes: {
+export default class ChangeAvatar extends React.Component {
+    static propTypes = {
         initialAvatarUrl: PropTypes.string,
         room: PropTypes.object,
         // if false, you need to call changeAvatar.onFileSelected yourself.
         showUploadSection: PropTypes.bool,
         width: PropTypes.number,
         height: PropTypes.number,
         className: PropTypes.string,
-    },
+    };
 
-    Phases: {
+    static Phases = {
         Display: "display",
         Uploading: "uploading",
         Error: "error",
-    },
-
-    getDefaultProps: function() {
-        return {
-            showUploadSection: true,
-            className: "",
-            width: 80,
-            height: 80,
-        };
-    },
+    };
+
+    static defaultProps = {
+        showUploadSection: true,
+        className: "",
+        width: 80,
+        height: 80,
+    };
+
+    constructor(props) {
+        super(props);
 
-    getInitialState: function() {
-        return {
+        this.state = {
             avatarUrl: this.props.initialAvatarUrl,
-            phase: this.Phases.Display,
+            phase: ChangeAvatar.Phases.Display,
         };
-    },
+    }
 
-    componentDidMount: function() {
+    componentDidMount() {
         MatrixClientPeg.get().on("RoomState.events", this.onRoomStateEvents);
-    },
+    }
 
     // TODO: [REACT-WARNING] Replace with appropriate lifecycle event
-    UNSAFE_componentWillReceiveProps: function(newProps) {
+    UNSAFE_componentWillReceiveProps(newProps) {
```

Comment:
Could we replace these with `getDerivedStateFromProps`? There are some more instances of this.

Reply:
Probably, but descoping fixing the UNSAFE_'s to minimize the risk

Reply:
makes sense